dashboard/app/views/devise/registrations/edit.html.haml

```diff
@@ -23,9 +25,12 @@
     - else
       .field
         = f.label :email
-        - if f.object.can_edit_email?
+        - if current_user.can_edit_email? && !current_user.oauth?
```

Comment:
In this context, no difference, `current_user` is passed as `f.object` in this particular scope; since this view now has logic outside that scope that needs access to that object, it seems better to be explicit and use `current_user`